packages/kbn-apm-synthtrace/src/lib/entities/entities_synthtrace_es_client.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { Client } from '@elastic/elasticsearch';
+import { EntityFields, ESDocumentWithOperation } from '@kbn/apm-synthtrace-client';
+import { pipeline, Readable, Transform } from 'stream';
+import { SynthtraceEsClient, SynthtraceEsClientOptions } from '../shared/base_client';
+import { getDedotTransform } from '../shared/get_dedot_transform';
+import { getSerializeTransform } from '../shared/get_serialize_transform';
+import { Logger } from '../utils/create_logger';
+
+export type EntitiesSynthtraceEsClientOptions = Omit<SynthtraceEsClientOptions, 'pipeline'>;
+
+export class EntitiesSynthtraceEsClient extends SynthtraceEsClient<EntityFields> {
+  constructor(options: { client: Client; logger: Logger } & EntitiesSynthtraceEsClientOptions) {
+    super({
+      ...options,
+      pipeline: entitiesPipeline(),
+    });
+    this.indices = ['.entities.v1.latest.builtin*'];
+  }
+}
+
+function entitiesPipeline() {
+  return (base: Readable) => {
+    return pipeline(
+      base,
+      getSerializeTransform(),
+      lastSeenTimestampTransform(),
+      getRoutingTransform(),
+      getDedotTransform(),
+      (err: unknown) => {
+        if (err) {
+          throw err;
+        }
+      }
+    );
+  };
+}
+
+function lastSeenTimestampTransform() {
+  return new Transform({
+    objectMode: true,
+    transform(document: ESDocumentWithOperation<EntityFields>, encoding, callback) {
+      const timestamp = document['@timestamp'];
+      if (timestamp) {
+        const isoString = new Date(timestamp).toISOString();
+        document['entity.lastSeenTimestamp'] = isoString;
+        document['event.ingested'] = isoString;
```

Comment:
[nit]: Do you always want to override this field ? Or only when its not present ?

Reply:
I always want to override it.